src/System.Globalization/tests/NumberFormatInfo/NumberFormatInfoData.cs

```diff
@@ -93,5 +97,22 @@ internal static int[] GetCurrencyNegativePatterns(CultureInfo cultureInfo)
 
             throw DateTimeFormatInfoData.GetCultureNotSupportedException(cultureInfo);
         }
+        
+        private static Version GetOSXKernelVersion()
+        {
+            if (s_isOSX)
+            {
+                ulong bytesLength = 256;
+                byte[] bytes = new byte[bytesLength];
+                sysctlbyname("kern.osrelease", bytes, ref bytesLength, null, 0);
+                string versionString = Encoding.UTF8.GetString(bytes);
+                return Version.Parse(versionString);
+            }
+            
+            return new Version(0, 0, 0);
+        }
+        
+        [DllImport("libc")]
+        private static extern int sysctlbyname(string ctlName, byte[] oldp, ref ulong oldpLen, byte[] newp, ulong newpLen);
```

Comment:
We don't need to write a regression test if we fixed the issue. that is my point, if we start writing the test way depend on the OS and on the data, we'll go into the same situation we have with the legacy tests. other example of this can be broken even if we test again some specific version and specific data is, someone on Windows install custom culture override the data. I know this scenario is not popular but possible.

by the way I wrote a test on Windows which do what I am saying and covering much more areas and was reliable when running on different Windows versions.

I see your point but I am seeing the heavy cost down the road too if we continue with the current approach. 

Also let's focus now on the original issue which is do we really need to provide the devs with aw ay to check specific OS type and version.